wolfssl/wolfcrypt/asn.h

```diff
@@ -513,7 +513,7 @@ WOLFSSL_LOCAL void SetASN_OID(ASNSetData *dataASN, int oid, int oidType);
  */
 #define SetASN_Buffer(dataASN, d, l)                                   \
     do {                                                               \
-        (dataASN)->data.buffer.data   = d;                             \
+        (dataASN)->data.buffer.data   = (const byte *) d;              \
```

Comment:
Revert this please.

Reply:
If I revert that, I get the following errors: 
```
In file included from wolfcrypt/src/asn.c:85:
wolfcrypt/src/asn.c: In function ‘SetEccPublicKey’:
./wolfssl/wolfcrypt/asn.h:516:39: error: assignment to ‘const byte *’ {aka ‘const unsigned char *’} from incompatible pointer type ‘const ecc_oid_t *’ {aka ‘const short unsigned int *’} [-Werror=incompatible-pointer-types]
  516 |         (dataASN)->data.buffer.data   = d;                             \
      |                                       ^
wolfcrypt/src/asn.c:21252:9: note: in expansion of macro ‘SetASN_Buffer’
21252 |         SetASN_Buffer(&dataASN[ECCPUBLICKEYASN_IDX_ALGOID_CURVEID],
      |         ^~~~~~~~~~~~~
  CC       wolfcrypt/src/src_libwolfssl_la-tfm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ecc.lo
wolfcrypt/src/asn.c: In function ‘wc_BuildEccKeyDer’:
./wolfssl/wolfcrypt/asn.h:516:39: error: assignment to ‘const byte *’ {aka ‘const unsigned char *’} from incompatible pointer type ‘const ecc_oid_t *’ {aka ‘const short unsigned int *’} [-Werror=incompatible-pointer-types]
  516 |         (dataASN)->data.buffer.data   = d;                             \
      |                                       ^
wolfcrypt/src/asn.c:28357:13: note: in expansion of macro ‘SetASN_Buffer’
28357 |             SetASN_Buffer(&dataASN[ECCKEYASN_IDX_CURVEID], key->dp->oid,
      |             ^~~~~~~~~~~~~
```

Reply:
Instead please add cast on the `(const byte*)key->dp->oid`. I'd like the macro to not have the cast in it to trap improper uses.

Reply:
Sure, I can do that. Thanks for the solution.